### PR TITLE
Display Taxes: ApplePay

### PR DESCRIPTION
**Pull request by deBhal**
#### Changes proposed in this Pull Request

This PR adds tax values to the ApplePay payment method. The backend is doing all the calculation, so all we need to do is check the `display_taxes` flag and then add the `total_tax` value (both from the cart).

I experimented with adding a "Subtotal" line, but the ApplePay API doesn't give us any formatting, and it ends up just looking cluttered and confusing so I took that out.

apple pay tax

Untitled_and_Checkout_‹_julesauspremiumtest_—_WordPress_com

#### Testing instructions

Testing this one will be tricky as you need to sandbox the back end to get tax values while also making Apple Pay happy.

There are a few tricks for testing Apple-Pay. The full write-up is at https://wp.me/p8hgLy-16d
 but I'll also note the minimum that I used to get it up and running:

- Set up an Apple Pay account. This will need a call to the bank to set it up for a MacBook. Alternatively, the link above has instructions for sandboxing the service to use a test account.
- Use a supported browser (Safari works, Chrome does not). Test your browser compatibility at https://applepaydemo.apple.com
- You'll need an https connection, so run calypso with `PROTOCOL=https npm run start`. (You can alternatively add it to your config, but https messes with debugging, so the environment variable seems more convenient).

### CODE REVIEW

**Comment by deBhal** [comment]
@Hywan: This looks right to me, but I haven't been able to make Applepay go through on the desktop, could you please review this extra-carefully and make sure you can actually buy something with it?

For me it pops up and then aborts after a second. The vendor verification seems to work, and I resolved a problem with sending the wrong environment to the backend, but I've run out of ideas to figure out what's going wrong after the `show()` - I'm happy to try and work this out with you, but I suspect that it's some weird corner case to do do with my half-legit/half-sandbox and not an actual problem.

File: client/my-sites/checkout/checkout/web-payment-box.jsx [hunk 1]
```diff
@@ -284,6 +296,22 @@ export class WebPaymentBox extends React.Component {
...
+				label: product.product_name,
+				amount: {
+					currency: product.currency,
+					value: product.cost,
+				},
+			};
+		} );
+		if ( config.isEnabled( 'show-tax' && total_tax ) ) {
+			displayItems = concat( displayItems, {
+				label: 'Tax',
```
Comment (lsl) [inline client/my-sites/checkout/checkout/web-payment-box.jsx]: Does this need translation?

**Comment by rralian** [comment]
@deBhal there was a bug on Stripe's side and it's finally been resolved. We have Apple Pay working in staging, but it's not launched yet. I don't think you'll be able to test it locally in calypso though because the domain where the request originates is important.

**Comment by deBhal** [comment]
Rebased.

@Hywan: Turns out the address problem was actually an authentication problem somehow. I'm pretty confident about this now, but I haven't managed to make a full purchase - would you mind doing that step and confirming there's not some surprise failure?

**Comment by DavidRothstein** [comment]
>I'm pretty confident about this now, but I haven't managed to make a full purchase - would you mind doing that step and confirming there's not some surprise failure?

I was able to go through and do a full test purchase and it worked correctly -- the correct tax was shown in Apple Pay and it matched what I was actually charged in the end.

**Comment by Hywan** [comment]
@DavidRothstein It means Apple Pay is deployed in production?

**Comment by DavidRothstein** [comment]
Hi @Hywan - hope you're doing well!

It's not quite deployed in production yet, although close.  I've only had time to work on it in bits and pieces, and we have a few things to try to fix (mainly with how things get displayed inside the Apple Pay payment sheet as well as how the credit card info gets displayed inside WordPress.com after a purchase), plus I found a bug where it didn't work with a particular type of credit card (easily fixed, but now want to test it more carefully to make sure there aren't other issues with other cards).  But we are hoping to get it out soon -- I will try to let you know when it's live :smiley:

**Comment by Hywan** [comment]
@DavidRothstein Let me know if I can hep!

**Pull request merged**